doc: nrf: Add nRF700X driver entry (#10781)
nrfconnect/sdk-nrf: doc/nrf/drivers/nrf700x.rst

.. _nrf700x_wifi:

nRF700X Wi-Fi Driver
####################

.. contents::
   :local:
   :depth: 2

This driver implements the Wi-Fi protocol for the nRF700X FullMAC series of devices. FullMAC devices implement the Wi-Fi
protocol in the chipset, and the driver only needs to configure the chipset and transfer the frames to and from the device

> to networking stack.

Reviewer: * Edited @richabp's suggestion for one additional improvement

> nRF700X is a companion chip and can be used with any SoC, primarily with Nordic Semiconductor's nRF53 and nRF91 Series SoCs.

Reviewer: Edited @richabp's suggestion for one additional improvement

> Driver can be enabled by using the :kconfig:option:`CONFIG_WIFI_NRF700X` Kconfig option.

Reviewer: Edited @richabp's suggestion for one additional improvement

Block diagram
*************

Below is the block diagram of the nRF700X Wi-Fi driver.

.. figure:: /images/nrf700x_wifi_driver.png
   :alt: nRF700X Wi-Fi driver block diagram
   :align: center
   :figclass: align-center

   nRF700X Wi-Fi driver block diagram

Design overview
***************

> The nRF700X Wi-Fi driver follows OS agnostic design, and the driver is implemented in to OS agnostic and OS (Zephyr) specific implementation files.

Reviewer: Edited @richabp's suggestion for one additional improvement

The OS agnostic files are located in :file:`drivers/wifi/nrf700x/osal/` folder, and the OS specific files are located in :file:`drivers/wifi/nrf700x/zephyr/` folder.

The driver is designed to be used with the Zephyr networking stack, and the driver is implemented as a network interface driver. Driver implements various IEEE082.11 Features
and supports the following features:

* Wi-Fi 6 (802.11ax) support
* WPA3/WPA2 personal security
* IEEE 802.11 power save modes
* Scan only mode
* IEEE 802.11 Station (STA) mode
* IEEE 802.11 SoftAP mode - not supported yet

> * Wi-Fi Direct mode - not supported yet

Author: Well, these are part of the driver code already, just that full end-end support is not added yet.

> * Wi-Fi Direct mode - not supported yet

Reviewer: Perhaps the last two can be added to its own list stating:

> The following features are in the driver code but not yet supported:
> * IEEE 802.11 SoftAP mode
> * Wi-Fi Direct mode

Except for scan  only mode, the driver uses `hostapd` to implement AP MLME and `wpa_supplicant` to implement 802.1X supplicant.

Driver to nRF700X communication
*******************************
The driver communicates with the nRF700X device using the QSPI/SPI interface. The driver uses the QSPI/SPI interface to send commands to the nRF700X device, and to transfer the
data to and from the device. nRF7002 DK uses QSPI where as nRF7002 EK uses SPI.

> For nRF7002 EK, a new shield `nrf7002_ek` is required to connect the nRF7002 EK to the SoC.

Reviewer: Edited @richabp's suggestion for one additional improvement

Radio test mode
***************
The nRF700X Wi-Fi driver supports radio test mode, which can be used to test the RF performance of the nRF700X device. This is a build time option, and can be enabled by
:kconfig:option:`CONFIG_NRF700X_RADIO_TEST` Kconfig option.

The radio test mode is implemented as a separate application, and the application is located in :file:`samples/wifi/radio_test/` folder.

API documentation
*****************

> | Source folder: :file:`drivers/wifi/nrf700x/`

Reviewer 2: Do we have any header file and source file to mention?

Author: there are too many, so, just pointed to the folder

After the nrf700x driver has been initialized, the application will see it as an Ethernet interface.

> To use that, the application can use `Zephyr Network APIs`_.

Reviewer: ```suggestion
To use the Ethernet interface, the application can use `Zephyr Network APIs`_.
```

.. _nrf700x_wifi:

nRF700X Wi-Fi driver
####################

.. contents::
   :local:
   :depth: 2

This driver implements the Wi-Fi® protocol for the nRF700X FullMAC series of devices.
FullMAC devices implement the Wi-Fi protocol in the chipset.
The driver configures the chipset and transfers the frames to and from the device to the networking stack.

nRF700X is a companion IC and can be used with any Nordic Semiconductor System-on-Chips (SoCs) such as the nRF53 and nRF91 Series SoCs.

You can enable the driver by using the :kconfig:option:`CONFIG_WIFI_NRF700X` Kconfig option.

Architecture
*************

The following figure illustrates the architecture of the nRF700X Wi-Fi driver.

.. figure:: /images/nrf700x_wifi_driver.svg
   :alt: nRF700X Wi-Fi driver block diagram
   :align: center
   :figclass: align-center

   nRF700X Wi-Fi driver architecture overview

Design overview
***************

> The nRF700X Wi-Fi driver follows a OS agnostic design, and the driver is implemented into OS agnostic and OS (Zephyr) specific implementation files.

Reviewer: ```suggestion
The nRF700X Wi-Fi driver follows an OS agnostic design, and the driver is implemented into OS agnostic and OS (Zephyr) specific implementation files.
```

The driver is designed to be used with the Zephyr networking stack.
It is implemented as a network interface driver.

The driver supports the following IEEE 802.11 features:

* Wi-Fi 6 (802.11ax) support
* WPA3/WPA2 personal security
* IEEE 802.11 power save modes
* Scan only mode
* IEEE 802.11 Station (STA) mode

The following features are in the driver code but not yet supported:

* IEEE 802.11 AP mode (Soft AP)
* Wi-Fi Direct mode

Except for scan only mode, the driver uses host access point daemon (hostapd) to implement AP Media Access Control (MAC) Sublayer Management Entity (AP MLME) and wpa_supplicant to implement 802.1X supplicant.

Driver to nRF700X communication
*******************************

The driver communicates with the nRF700X device using the QSPI/SPI interface.
The driver uses the QSPI/SPI interface to send commands to the nRF700X device, and to transfer the data to and from the device.
The nRF7002 DK uses QSPI whereas the nRF7002 EK uses SPI.

To connect the nRF7002 EK to the SoC, the ``nrf7002_ek`` shield is required.

> Radio test mode

Reviewer 2: ```suggestion
Radio Test mode
```

***************

> The nRF700X Wi-Fi driver supports radio test mode, which can be used to test the RF performance of the nRF700X device.

Reviewer 2: ```suggestion
The nRF700X Wi-Fi driver supports Radio Test mode, which can be used to test the RF performance of the nRF700X device.
```

This is a build time option and can be enabled by using the :kconfig:option:`CONFIG_NRF700X_RADIO_TEST` Kconfig option.

> For more details about using this driver in radio test mode, see :ref:`wifi_radio_test`.

Reviewer 2: ```suggestion
For more details about using this driver in Radio Test mode, see :ref:`wifi_radio_test`.
```
